src/OpenDebugAD7/AD7DebugSession.cs

```diff
@@ -2043,11 +2044,24 @@ protected override void HandleThreadsRequestAsync(IRequestResponder<ThreadsArgum
             }
 
             // iterate over the collection asking the engine for the name
-            foreach (var pair in threads)
+            foreach ((int id, IDebugThread2 thread) in threads)
             {
-                string name;
-                pair.Value.GetName(out name);
-                response.Threads.Add(new OpenDebugThread(pair.Key, name));
+                string name = null;
+
+                if (thread != null && thread.GetName(out name) == HRConstants.S_OK)
```

Comment:
>thread.GetName(out name) == HRConstants.S_OK [](http://example.com/codeflow?start=38&length=44)

Seems like we should either:
1. Treat GetName failures as fatal for that thread, and ignore the thread.
-or-
2. Treat GetName failures the same as if string.Empty was returned, and use a name of `[<thread-id-here>]`

Reply:
This will be handled as #2 in the OpenDebugThread class.

See https://github.com/microsoft/MIEngine/blob/496f56bbed40ab22f3cbaea58cb7e6ebf18f310c/src/OpenDebugAD7/OpenDebug/CustomProtocolObjects.cs#L75-L78